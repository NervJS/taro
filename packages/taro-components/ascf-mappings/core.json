{
  "Button": {
    "size": {
      "ascfProp": "dimension",
      "valueMap": {
        "mini": "XS",
        "small": "S",
        "default": "M",
        "large": "L"
      },
      "required": false,
      "default": "M",
      "docLink": "https://ascf.docs/components/button#dimension"
    },
    "loading": {
      "ascfProp": "isLoading",
      "type": "boolean",
      "required": false,
      "default": false,
      "docLink": "https://ascf.docs/components/button#isloading"
    }
  },
  "Input": {
    "password": {
      "ascfProp": "secretInput",
      "type": "boolean",
      "required": false,
      "default": false,
      "docLink": "https://ascf.docs/components/input#secretinput"
    }
  }
}
