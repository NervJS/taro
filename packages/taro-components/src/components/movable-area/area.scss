taro-movable-area-core {
  display: block;
  position: relative;
  width: 10px;
  height: 10px;
}
