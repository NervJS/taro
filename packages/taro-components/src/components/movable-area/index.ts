export * from './movable-area'
export * from './movable-view'
