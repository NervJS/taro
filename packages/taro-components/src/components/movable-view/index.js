import Nerv from 'nervjs'

class MovableView extends Nerv.Component {
  componentDidMount () {
    console.error('H5 暂不支持 MovableView 组件！')
  }

  render () {
    const { ...reset } = this.props
    return <div {...reset}>{this.props.children}</div>
  }
}

class MovableArea extends Nerv.Component {
  componentDidMount () {
    console.error('H5 暂不支持 MovableArea 组件！')
  }

  render () {
    const { ...reset } = this.props
    return <div {...reset}>{this.props.children}</div>
  }
}

export { MovableArea, MovableView }
