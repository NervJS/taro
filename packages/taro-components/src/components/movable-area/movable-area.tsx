import { Component, h, Host, Prop, Element } from '@stencil/core'

@Component({
  tag: 'taro-movable-area-core',
  styleUrl: './area.scss'
})
export class MovableArea {

  /** 
   * 当里面的 movable-view 设置为支持双指缩放时，设置此值可将缩放手势生效区域修改为整个movable-area
   */
  @Prop() scaleArea: boolean

  @Element() element: HTMLElement

  /** 观察者 */
  private observer?: MutationObserver
  /** 子元素集合 */
  private views: Array<HTMLElement> = []
  /** 单个元素缩放时的目标元素 */
  private scaleTarget?: HTMLElement
  /** 手势缩放 */
  private scaleLength: number = 0
  /** 宽高 */
  private offset?: { width: number; height: number; }

  connectedCallback () {
    this.observer = new MutationObserver((mutations) => {
      mutations.forEach((mutation) => {
        if (mutation.attributeName === "class" || mutation.attributeName === "style") {
          const offsetWidth = this.element.offsetWidth
          const offsetHeight = this.element.offsetHeight
          if (offsetWidth !== this.offset?.width || offsetHeight !== this.offset?.height) {
            this.updateArea()
          }
          this.offset = {
            width: offsetWidth,
            height: offsetHeight
          }
        }
      })
    })

    this.observer.observe(this.element, {
      attributes: true
    })
  }

  disconnectedCallback () {
    this.observer?.disconnect()
  }

  componentDidLoad () {
    this.viewsChanged()
  }

  viewsChanged = () => {
    this.views = []
    const elements = this.element.querySelectorAll('taro-movable-view-core')
    Array.from(elements).forEach((element) => {
      this.views.push(element)
    })
    this.updateArea()
  }

  handleTouchStart = (e: TouchEvent) => {
    const touches = e.touches
    if (!touches || touches.length <= 1) {
      return
    }

    const gap = {
      width: touches[1].pageX - touches[0].pageX,
      height: touches[1].pageY - touches[0].pageY
    }
    this.scaleLength = Math.sqrt(gap.width * gap.width + gap.height * gap.height)
    if (this.scaleArea) {
      return
    }

    const find = (target: EventTarget, views: Array<HTMLElement>) => {
      const loop = (e, t) => {
        if (!(e = e.parentNode)) {
          return false
        }
        return (!(e instanceof HTMLElement) || e !== document.body) && (e === t || e === t.element || e.element === t || loop(e, t))
      }

      for (let i = 0; i < views.length; i++) {
        const view = views[i]
        if (target === view["element"] || loop(target, view)) {
          return view
        }
      }
    }

    const touch1 = find(touches[0].target, this.views)
    const touch2 = find(touches[1].target, this.views)
    this.scaleTarget = touch1 && touch1 === touch2 ? touch1 : undefined
  }

  handleTouchMove = (e: TouchEvent) => {
    const touches = e.touches
    if (!touches || touches.length <= 1) {
      return
    }
    e.preventDefault()
    const gap = {
      width: touches[1].pageX - touches[0].pageX,
      height: touches[1].pageY - touches[0].pageY
    }
    if (this.scaleLength > 0) {
      this.updateScale(Math.sqrt(gap.width * gap.width + gap.height * gap.height) / this.scaleLength)
    }
  }

  handleTouchEnd = (e: TouchEvent) => {
    if ((e.touches && e.touches.length) || !e.changedTouches) {
      return
    }
    this.scaleLength = 0
    if (this.scaleArea) {
      this.views.forEach((element) => {
        element["endScale"]?.()
      })
    } else {
      this.scaleTarget?.["endScale"]?.()
    }
    this.scaleTarget = undefined
  }

  updateScale = (scale: number) => {
    if (!scale || scale === 1) {
      return
    }

    if (this.scaleArea) {
      this.views.forEach((element) => {
        element["setScale"]?.(scale)
      })
    } else {
      this.scaleTarget?.["setScale"]?.(scale)
    }
  }

  updateArea = () => {
    const computedStyle = window.getComputedStyle(this.element)
    const clientRect = this.element.getBoundingClientRect()
    const horizontal = ["Left", "Right"].map((e) => {
      return parseFloat(computedStyle["border" + e + "Width"]) + parseFloat(computedStyle["padding" + e])
    })
    const vertical = ["Top", "Bottom"].map((e) => {
      return parseFloat(computedStyle["border" + e + "Width"]) + parseFloat(computedStyle["padding" + e])
    })

    this.views.forEach((element) => {
      element["setParent"]?.({
        element: this.element,
        area: {
          height: clientRect.height - vertical[0] - vertical[1],
          width: clientRect.width - horizontal[0] - horizontal[1]
        }
      })
    })
  }

  render () {
    return (
      <Host
        onTouchStart={this.handleTouchStart}
        onTouchMove={this.handleTouchMove}
        onTouchEnd={this.handleTouchEnd}
      />
    )
  }
}
