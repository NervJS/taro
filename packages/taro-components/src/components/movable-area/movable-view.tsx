import { Component, h, Host, Prop, Event, EventEmitter, Element, Watch, Method } from '@stencil/core'

@Component({
  tag: 'taro-movable-view-core',
  styleUrl: './view.scss'
})
export class MovableView {
  /**
   * 定义x轴方向的偏移，如果x的值不在可移动范围内，会自动移动到可移动范围；改变x的值会触发动画；单位支持px；
   */
  @Prop() x: number | string = 0
  /**
   * 定义y轴方向的偏移，如果y的值不在可移动范围内，会自动移动到可移动范围；改变y的值会触发动画；单位支持px；
   */
  @Prop() y: number | string = 0
  /**
   * 移动方向，属性值有all、vertical、horizontal、none
   */
  @Prop() direction: "all" | "vertical" | "horizontal" | "none" = "none"
  /**
   * 超过可移动区域后，是否还可以移动
   */
  @Prop() outOfBounds: boolean = false
  /** 
   * 是否带有惯性 
   */
  @Prop() inertia: boolean = false
  /**
   * 摩擦系数，用于控制惯性滑动的动画，值越大摩擦力越大，滑动越快停止；必须大于0，否则会被设置成默认值
   */
  @Prop() friction: number = 2
  /**
   * 阻尼系数，用于控制x或y改变时的动画和过界回弹的动画，值越大移动越快
   */
  @Prop() damping: number = 20
  /**
   * 是否禁用
   */
  @Prop() disabled: boolean = false
  /**
   * 是否支持双指缩放，默认缩放手势生效区域是在movable-view内
   */
  @Prop() scale: boolean = false
  /**
   * 定义缩放倍数最小值
   */
  @Prop() scaleMin: number = .5
  /**
   * 定义缩放倍数最大值
   */
  @Prop() scaleMax: number = 10
  /**
   * 定义缩放倍数，取值范围为 0.5 - 10
   */
  @Prop() scaleValue: number = 1
  /**
   * 是否使用动画
   */
  @Prop() animation: boolean = true

  /**
   * 拖动过程中触发的事件，event.detail = {x, y, source}
   */
  @Event({
    eventName: 'change'
  }) onChange: EventEmitter

  /**
   * 缩放过程中触发的事件，event.detail = {x, y, scale}，x和y字段在2.1.0之后支持
   */
  @Event({
    eventName: 'scale'
  }) onScale: EventEmitter

  /**
   * 初次手指触摸后移动为横向的移动时触发，如果catch此事件，则意味着touchmove事件也被catch
   */
  @Event({
    eventName: 'htouchmove'
  }) onHTouchMove: EventEmitter

  /**
   * 初次手指触摸后移动为纵向的移动时触发，如果catch此事件，则意味着touchmove事件也被catch
   */
  @Event({
    eventName: 'vtouchmove'
  }) onVTouchMove: EventEmitter

  @Element() element: HTMLElement

  @Watch('x')
  watchX (newValue: number | string) {
    this.setTransform(parseFloat(`${newValue || 0}`), this.translateY)
  }

  @Watch('y')
  watchY (newValue: number | string) {
    this.setTransform(this.translateX, parseFloat(`${newValue || 0}`))
  }

  @Watch('scaleMin')
  @Watch('scaleMax')
  watchScaleMinOrMax () {
    if (!this.scale) return false
    this.updateScale(this.currentScale, true)
    this.setOriginScale(this.currentScale)
  }

  @Watch('scaleValue')
  watchScaleValue (scale) {
    if (!this.scale) {
      return false
    }
    this.updateScale(scale, true)
    this.setOriginScale(scale)

    return scale
  }

  /**
   * 设置父节点
   */
  @Method()
  async setParent ({ element, area }: { element: HTMLElement, area: { width: number; height: number; } }) {
    const scale = this.scale ? this.scaleValue : 1
    this.area = area
    this.parent = element

    this.updateOffset()
    this.updateScaleOffset(scale)
    this.updateBoundary()
    this.setTransform(Number(this.x) + this.scaleOffset.x, Number(this.y) + this.scaleOffset.y, scale, "", true)
    this.setOriginScale(scale)
  }

  /**
   * 结束缩放
   */
  @Method()
  async endScale () {
    this.scaling = false
    this.setOriginScale(this.currentScale)
  }

  /**
   * 更新缩放
   */
  @Method()
  async setScale (scale: number) {
    if (!this.scale) {
      return
    }
    this.scaling = true
    this.updateScale(scale * this.originScale)
  }

  connectedCallback () {
    this.observer = new MutationObserver((mutations) => {
      mutations.forEach((mutation) => {
        const name = mutation.attributeName
        if (name && ["class", "style"].includes(name)) {
          const oldValue = mutation.oldValue
          const newValue = (mutation.target as HTMLElement).getAttribute(name)
          if (oldValue === newValue) {
            return
          }
          const filter = (input: string | null) => {
            return input?.split(";").filter((item) => {
              return !["transform", "will-change"].find((key) => {
                return item.trim().startsWith(key)
              })
            }).join(";")
          }
          if (name === "style" && filter(newValue) === filter(oldValue)) {
            return
          }

          this.updateOffset()
          this.updateScaleOffset()
          this.updateBoundary()
          this.setTransform(this.translateX, this.translateY)
        }
      })
    })

    this.observer.observe(this.element, {
      attributes: true,
      attributeOldValue: true
    })
  }

  disconnectedCallback () {
    this.observer?.disconnect()
  }

  /** 观察者 */
  private observer?: MutationObserver
  /** 缩放中 */
  private scaling: boolean
  /** 移动中 */
  private touching: boolean
  /** 更新中 */
  private updating: boolean
  /** 水平方向是否可移动 */
  private xMove: boolean
  /** 垂直方向是否可移动 */
  private yMove: boolean
  /** 首次移动方向事件是否已触发 */
  private firstMoveFireEvent: boolean
  /** 当前水平偏移 */
  private translateX: number = 0
  /** 当前垂直偏移 */
  private translateY: number = 0
  /** touch-start 原点 */
  private origin: { x: number; y: number; } = { x: 0, y: 0 }
  /** 父容器大小 */
  private area: { width: number; height: number; } = { width: 0, height: 0 }
  /** 父容器 */
  private parent?: HTMLElement
  /** 原始缩放倍数 */
  private originScale: number = 1
  /** 当前缩放倍数 */
  private currentScale: number = 1
  /** 宽度 */
  private width: number = 0
  /** 高度 */
  private height: number = 0
  /** 移动边界 */
  private minX: number = 0
  private minY: number = 0
  private maxX: number = 0
  private maxY: number = 0
  /** 移动基础位置 */
  private baseX: number = 0
  private baseY: number = 0
  /** 偏移量 */
  private offset: { x: number; y: number; } = { x: 0, y: 0 }
  private scaleOffset: { x: number; y: number; } = { x: 0, y: 0 }

  getLimitXY = (x: number, y: number) => {
    let outOfBounds = false
    x > this.maxX ? (x = this.maxX, outOfBounds = true) : x < this.minX && (x = this.minX, outOfBounds = true)
    y > this.maxY ? (y = this.maxY, outOfBounds = true) : y < this.minY && (y = this.minY, outOfBounds = true)

    return { x, y, outOfBounds }
  }

  animationTo = (x: number, y: number, scale?: number, source?: string, noEmitChange?: boolean, emitScale?: boolean, callback?: () => void) => {
    if (this.animation) {
      this.setTransform(x, y, scale, source, noEmitChange, emitScale)
      callback?.()
    } else {
      this.setTransform(x, y, scale, source, noEmitChange, emitScale)
    }
  }

  setTransform = (x: number, y: number, scale?: number, source?: string, noEmitChange?: boolean, emitScale?: boolean) => {
    x = Number(x.toFixed(1))
    y = Number(y.toFixed(1))
    scale = Number((scale ?? this.currentScale).toFixed(3))

    if (!this.outOfBounds) {
      const limit = this.getLimitXY(x, y)
      x = limit.x
      y = limit.y
    }

    const subtract = (e, t) => {
      return +((1e3 * e - 1e3 * t) / 1e3).toFixed(1)
    }

    const realX = subtract(x, this.scaleOffset.x)
    const realY = subtract(y, this.scaleOffset.y)
    if (this.translateX !== x || this.translateY !== y) {
      !noEmitChange && this.onChange.emit({
        x: realX,
        y: realY,
        source
      })
    }

    if (scale !== this.currentScale) {
      emitScale && this.onScale.emit({
        scale,
        x: realX,
        y: realY
      })
    }

    const transform = `translateX(${x}px) translateY(${y}px) translateZ(0px) scale(${scale})`
    this.element.style.transform = transform
    this.element.style.webkitTransform = transform
    this.translateX = x
    this.translateY = y
    this.currentScale = scale
  }

  updateOffset = () => {
    const offset = (element: HTMLElement, parent: HTMLElement): { left: number; top: number; } => {
      if (element === parent || !element.offsetParent) {
        return { left: 0, top: 0 }
      }
      const current = offset(element.offsetParent as HTMLElement, parent)
      return {
        left: element.offsetLeft + current.left,
        top: element.offsetTop + current.top
      }
    }

    if (!this.parent) {
      return
    }

    const current = offset(this.element, this.parent)

    this.offset.x = current.left
    this.offset.y = current.top
  }

  updateScaleOffset = (scale: number = this.currentScale) => {
    const rect = this.element.getBoundingClientRect()
    this.height = rect.height / this.currentScale
    this.width = rect.width / this.currentScale
    this.scaleOffset.x = (this.width * scale - this.width) / 2
    this.scaleOffset.y = (this.height * scale - this.height) / 2
  }

  updateBoundary = () => {
    const x1 = 0 - this.offset.x + this.scaleOffset.x
    const x2 = this.area.width - this.width - this.offset.x - this.scaleOffset.x
    this.minX = Math.min(x1, x2)
    this.maxX = Math.max(x1, x2)
    const y1 = 0 - this.offset.y + this.scaleOffset.y
    const y2 = this.area.height - this.height - this.offset.y - this.scaleOffset.y
    this.minY = Math.min(y1, y2)
    this.maxY = Math.max(y1, y2)
  }

  updateScale = (scale: number, animation?: boolean, animationCallback?: () => void) => {
    if (!this.scale) {
      return
    }

    const target = this.adjustScale(scale)
    this.updateScaleOffset(target)
    this.updateBoundary()

    const { x, y } = this.getLimitXY(this.translateX, this.translateY)
    if (animation) {
      this.animationTo(x, y, target, "", true, true, animationCallback)
    } else if (!this.updating) {
      this.updating = true
      requestAnimationFrame(() => {
        this.setTransform(x, y, target, "", true, true)
        this.updating = false
      })
    }
  }

  setOriginScale = (scale: number) => {
    this.originScale = scale
  }

  adjustScale = (scale: number) => {
    return Math.min(10, this.scaleMax, Math.max(.5, this.scaleMin, scale))
  }

  componentDidLoad () {
    this.element.style.transformOrigin = "center"

    this.xMove = ["horizontal", "all"].includes(this.direction)
    this.yMove = ["vertical", "all"].includes(this.direction)
    if (this.friction <= 0) {
      this.friction = 2
    }

    if (this.x || this.y) {
      const x = parseFloat(`${this.x || 0}`)
      const y = parseFloat(`${this.y || 0}`)
      this.setTransform(x, y)
    }
  }

  handleTouchStart = (e: TouchEvent) => {
    const touches = e.touches
    if (this.disabled || touches.length > 1 || !this.element) {
      return
    }
    const touch = touches[0]

    this.touching = true
    this.firstMoveFireEvent = false
    this.origin.x = touch.screenX
    this.origin.y = touch.screenY

    this.baseX = this.translateX
    this.baseY = this.translateY

    this.element.style.willChange = "transform"
  }

  handleTouchMove = (e: TouchEvent) => {
    const touches = e.touches
    if (this.disabled || !this.element || this.scaling || !this.touching || touches.length > 1) {
      return
    }
    e.preventDefault()

    const touch = touches[0]

    const x = touch.screenX - this.origin.x
    const y = touch.screenY - this.origin.y

    this.setTransform(this.xMove ? (x + this.baseX) : 0, this.yMove ? (y + this.baseY) : 0)

    if (!this.firstMoveFireEvent) {
      this.firstMoveFireEvent = true
      const onTouchMove = Math.abs(x) > Math.abs(y) ? this.onHTouchMove : this.onVTouchMove
      onTouchMove.emit({
        originalEvent: e,
        bubbles: false,
        capturePhase: false,
        composed: true,
        extraFields: {
          touches: e.touches || {},
          changedTouches: e.changedTouches || {}
        }
      })
    }
  }

  handleTouchEnd = (e: TouchEvent) => {
    const touch = e.changedTouches[0]
    if (this.disabled || !this.touching || !touch) {
      return
    }
    this.touching = false

    const x = touch.screenX - this.origin.x
    const y = touch.screenY - this.origin.y

    this.setTransform(this.xMove ? (x + this.baseX) : 0, this.yMove ? (y + this.baseY) : 0)
  }

  render () {
    return (
      <Host
        onTouchStart={this.handleTouchStart}
        onTouchMove={this.handleTouchMove}
        onTouchEnd={this.handleTouchEnd}
      />
    )
  }
}
