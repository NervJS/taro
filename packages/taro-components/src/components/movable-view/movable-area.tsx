// eslint-disable-next-line @typescript-eslint/no-unused-vars
import { Component, h, Host } from '@stencil/core'

@Component({
  tag: 'taro-movable-area-core'
})
export class MovableArea {
  componentDidLoad () {
    console.error('H5 暂不支持 MovableArea 组件！')
  }

  render () {
    return (
      <Host />
    )
  }
}
