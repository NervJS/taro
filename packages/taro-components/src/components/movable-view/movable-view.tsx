// eslint-disable-next-line @typescript-eslint/no-unused-vars
import { Component, h, Host } from '@stencil/core'

@Component({
  tag: 'taro-movable-view-core'
})
export class MovableView {
  componentDidLoad () {
    console.error('H5 暂不支持 MovableView 组件！')
  }

  render () {
    return (
      <Host />
    )
  }
}
