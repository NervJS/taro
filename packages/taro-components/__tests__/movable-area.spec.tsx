import { h } from '@stencil/core'
import { newSpecPage, SpecPage } from '@stencil/core/testing'

import { MovableArea } from '../src/components/movable-area/movable-area'
import { MovableView } from '../src/components/movable-area/movable-view'

describe('MovableArea', () => {
  let page: SpecPage

  it('base', async () => {
    page = await newSpecPage({
      components: [MovableArea, MovableView],
      template: () => (<taro-movable-area-core style='height: 200px; width: 200px; background: red;'>
        <taro-movable-view-core style='height: 50px; width: 50px; background: blue;' direction='all'>
          带我走
        </taro-movable-view-core>
      </taro-movable-area-core>),
    })
    expect(page.root).toMatchSnapshot()
  })
})
