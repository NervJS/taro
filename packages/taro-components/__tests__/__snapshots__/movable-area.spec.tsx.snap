// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovableArea base 1`] = `
<taro-movable-area-core style="0: h; 1: e; 2: i; 3: g; 4: h; 5: t; 6: :; 7:  ; 8: 2; 9: 0; 10: 0; 11: p; 12: x; 13: ;; 14:  ; 15: w; 16: i; 17: d; 18: t; 19: h; 20: :; 21:  ; 22: 2; 23: 0; 24: 0; 25: p; 26: x; 27: ;; 28:  ; 29: b; 30: a; 31: c; 32: k; 33: g; 34: r; 35: o; 36: u; 37: n; 38: d; 39: :; 40:  ; 41: r; 42: e; 43: d; 44: ;;">
  <taro-movable-view-core style="0: h; 1: e; 2: i; 3: g; 4: h; 5: t; 6: :; 7:  ; 8: 5; 9: 0; 10: p; 11: x; 12: ;; 13:  ; 14: w; 15: i; 16: d; 17: t; 18: h; 19: :; 20:  ; 21: 5; 22: 0; 23: p; 24: x; 25: ;; 26:  ; 27: b; 28: a; 29: c; 30: k; 31: g; 32: r; 33: o; 34: u; 35: n; 36: d; 37: :; 38:  ; 39: b; 40: l; 41: u; 42: e; 43: ;; transform-origin: center; transform: translateX(0px) translateY(0px) translateZ(0px) scale(1); webkit-transform: translateX(0px) translateY(0px) translateZ(0px) scale(1);">
    带我走
  </taro-movable-view-core>
</taro-movable-area-core>
`;
